docs: add section about fake user data (#154065)
NixOS/nixpkgs: doc/contributing/coding-conventions.chapter.md

- `https://` URLs are not secure in method 5. When obtaining hashes with fake hash method, TLS checks are disabled. So refetch source hash from several different networks to exclude MITM scenario. Alternatively, use fake hash method to make Nix error, but instead of extracting hash from error, extract `https://` URL and prefetch it with method 1.

## Using fake user data for download {#sec-fake-data}

> Some software require user data like name and address to download it. When it does not matter which data we put in, we can use this fake data.

Reviewer: ```suggestion
Some software downloads require user data, such as name and/or address. When possible, the following data should be used:
```

Fixes grammar
